Kernel/Syscalls/sched.cpp

```diff
@@ -17,58 +19,77 @@ ErrorOr<FlatPtr> Process::sys$yield()
     return 0;
 }
 
-ErrorOr<FlatPtr> Process::sys$sched_setparam(int pid, Userspace<const struct sched_param*> user_param)
+ErrorOr<NonnullRefPtr<Thread>> Process::get_thread_from_pid_or_tid(pid_t pid_or_tid, Syscall::scheduler_parameters_mode mode)
+{
+    VERIFY(g_scheduler_lock.is_locked_by_current_processor());
+    Thread* peer;
+    if (mode == Syscall::scheduler_parameters_mode::THREAD) {
+        peer = Thread::current();
+        if (pid_or_tid != 0)
+            peer = Thread::from_tid(pid_or_tid);
+
+        // Syscall::scheduler_parameters_mode::PROCESS
```

Comment:
If this is really only intended for `::Process`, maybe make that explicit and add an `else { VERIFY_NOT_REACHED(); }`

Reply:
It's not? The else is perfectly functional and the syscall should be able to handle both processes and threads. The implementation is just not there yet.

Reply:
No, I meant, if the current else-branch really only applies for Process queries (and not for any non-Thread query, which is what I assume this comment implies), this should look like the following:

```
if (mode == Syscall::SchedulerParametersMode::Thread) {
    // some code
} else if (mode == Syscall::SchedulerParametersMode::Process) {
    // more code
} else {
    VERIFY_NOT_REACHED();
}
```

This both guards against the case of us forgetting something if we ever add a third mode, and it make sure that we actually don't allow any mode, but just the two that we defined (although, there are certainly better methods for doing that than just hitting an assert in the kernel).
```

Reply:
This looks like it was resolved mistakenly, at least in my opinion.